api/saml/parser.py

```diff
@@ -467,3 +467,89 @@ def parse(self, xml_metadata):
         self._logger.info('Finished parsing an XML string containing SAML metadata')
 
         return providers
+
+
+class SAMLSubjectParser(object):
+    """Parses SAML response into Subject object"""
+
+    def _parse_name_id(self, name_id_attributes):
+        """Parses NameID attributes
+
+        :param name_id_attributes: Dictionary containing NameID attributes
+        :type name_id_attributes: Dict
+
+        :return: NameID object
+        :rtype: NameID
+        """
+        name_id = NameID(
+            name_id_attributes['Format'],
+            name_id_attributes['NameQualifier'],
+            None,
+            name_id_attributes['value']
+        )
+
+        return name_id
+
+    def _parse_attribute_values(self, attribute_values):
+        """Parses attribute values
+
+        :param attribute_values: List containing SAML attribute values
+        :type attribute_values: List[Union[str, Dict]]
+
+        :return: 2-tupple containing a NameID and a list of SAML attribute values
+        :rtype: Tuple[Optional[NameID], List[str]]
```

Comment:
This docstring doesn't match the code; you're only returning the list of attribute values.

Reply:
Updated it, thank you